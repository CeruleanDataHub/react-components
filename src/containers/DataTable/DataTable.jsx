import PropTypes from "prop-types";
import React from "react";
import DataTableWrapper from "react-data-table-component";

export const DataTable = ({ columns, data }) => (
  <DataTableWrapper
    pagination={data.length > 10}
    striped
    highlightOnHover
    noHeader
    columns={columns}
    data={data}
  />
);

DataTable.propTypes = {
  /** Table columns */
  // eslint-disable-next-line react/forbid-prop-types
  columns: PropTypes.array,
  /** Table data. List of objects. You can create create a custom cell containing an react component by adding an `cell` value to the object */
  // eslint-disable-next-line react/forbid-prop-types
  data: PropTypes.array
};

DataTable.defaultProps = {
  columns: [],
  data: []
};
