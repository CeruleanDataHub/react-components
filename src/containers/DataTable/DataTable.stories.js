import React from "react";

import { DataTable } from "./DataTable";

export default {
  title: "DataTable",
  component: DataTable
};

const data = [
  { id: 1, title: "Conan the Barbarian", year: "1982" },
  { id: 2, title: "Conan the Barbarian 2", year: "1984" },
  { id: 3, title: "Conan the Barbarian 3", year: "1985" }
];

const columns = [
  { id: 1, name: "Title", selector: "title" },
  { id: 2, name: "Year", selector: "year" }
];

export const DataTableStory = () => <DataTable columns={columns} data={data} />;

export const DataTableCustomCellStory = () => {
  const columns2 = [
    {
      name: "Custom Title",
      cell: row => <div>{row.title}</div>
    },
    {
      name: "Fancy title",
      cell: row => (
        <div style={{ color: "red", fontSize: "1.2rem" }}>{row.title}</div>
      )
    },
    {
      name: "Link",
      cell: row => (
        <a href={row.id} target="_blank" rel="noopener noreferrer">
          {/* eslint-disable-next-line react/jsx-one-expression-per-line */}
          This is a link to {row.title}
        </a>
      )
    }
  ];
  return <DataTable columns={columns2} data={data} />;
};
