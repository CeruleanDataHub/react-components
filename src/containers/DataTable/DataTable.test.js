import React from "react";
import { mount } from "enzyme";

import { DataTable } from "./DataTable";

describe("DataTable", () => {
  let component;
  let columnsStub;
  let dataStub;

  beforeEach(() => {
    dataStub = [
      { id: 1, title: "Conan the Barbarian", year: "1982" },
      { id: 2, title: "Conan the Barbarian 2", year: "1984" },
      { id: 3, title: "Conan the Barbarian 3", year: "1985" }
    ];

    columnsStub = [
      { id: 1, name: "Title", selector: "title" },
      { id: 2, name: "Year", selector: "year" }
    ];

    component = mount(<DataTable columns={columnsStub} data={dataStub} />);
  });

  it("renders", () => {
    expect(component).toMatchHtmlSnapshot();
  });

  it("given big amount of data, renders with pagination", () => {
    dataStub = [
      { id: 1, title: "Conan the Barbarian", year: "1982" },
      { id: 2, title: "Conan the Barbarian 2", year: "1984" },
      { id: 3, title: "Conan the Barbarian 3", year: "1985" },
      { id: 4, title: "Conan the Barbarian", year: "1982" },
      { id: 5, title: "Conan the Barbarian 2", year: "1984" },
      { id: 6, title: "Conan the Barbarian 3", year: "1985" },
      { id: 7, title: "Conan the Barbarian", year: "1982" },
      { id: 8, title: "Conan the Barbarian 2", year: "1984" },
      { id: 9, title: "Conan the Barbarian 3", year: "1985" },
      { id: 10, title: "Conan the Barbarian", year: "1982" },
      { id: 11, title: "Conan the Barbarian 2", year: "1984" }
    ];

    columnsStub = [
      { id: 1, name: "Title", selector: "title" },
      { id: 2, name: "Year", selector: "year" }
    ];

    component = mount(<DataTable columns={columnsStub} data={dataStub} />);

    expect(component).toMatchHtmlSnapshot();
  });
});
