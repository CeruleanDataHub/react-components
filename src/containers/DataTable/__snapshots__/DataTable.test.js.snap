// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable should render 1`] = `
<div
  className="sc-fznxsB forRdO"
  style={Object {}}
>
  <div
    className="sc-fznWqX gnahTY"
  >
    <div
      className="sc-AxjAm gIMaKV rdt_Table"
      disabled={false}
      role="table"
    >
      <div
        className="sc-AxirZ eBuPmJ rdt_TableHead"
        role="rowgroup"
      >
        <div
          className="sc-AxiKw cTEQON rdt_TableHeadRow"
          disabled={false}
          role="row"
        >
          <div
            className="sc-AxhCb sc-AxhUy sc-fzplWN HWHoi rdt_TableCol"
          >
            <div
              className="sc-fznyAO cxtWCz rdt_TableCol_Sortable"
              id="column-title"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="button"
              tabIndex={0}
            >
              <div>
                Title
              </div>
            </div>
          </div>
          <div
            className="sc-AxhCb sc-AxhUy sc-fzplWN HWHoi rdt_TableCol"
          >
            <div
              className="sc-fznyAO cxtWCz rdt_TableCol_Sortable"
              id="column-year"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="button"
              tabIndex={0}
            >
              <div>
                Year
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="sc-fzpjYC ieTOoW rdt_TableBody"
        offset="250px"
        role="rowgroup"
      >
        <div
          className="sc-fzoLsD jekBnA rdt_TableRow"
          id="row-1"
          onClick={[Function]}
          onDoubleClick={[Function]}
          role="row"
          selected={false}
        >
          <div
            className="sc-AxhCb sc-AxhUy sc-AxgMl gbbhfF rdt_TableCell"
            data-tag="___react-data-table-allow-propagation___"
            id="cell-1-1"
            role="cell"
          >
            <div
              data-tag="___react-data-table-allow-propagation___"
            >
              Conan the Barbarian
            </div>
          </div>
          <div
            className="sc-AxhCb sc-AxhUy sc-AxgMl gbbhfF rdt_TableCell"
            data-tag="___react-data-table-allow-propagation___"
            id="cell-2-1"
            role="cell"
          >
            <div
              data-tag="___react-data-table-allow-propagation___"
            >
              1982
            </div>
          </div>
        </div>
        <div
          className="sc-fzoLsD jekBnA rdt_TableRow"
          id="row-2"
          onClick={[Function]}
          onDoubleClick={[Function]}
          role="row"
          selected={false}
        >
          <div
            className="sc-AxhCb sc-AxhUy sc-AxgMl gbbhfF rdt_TableCell"
            data-tag="___react-data-table-allow-propagation___"
            id="cell-1-2"
            role="cell"
          >
            <div
              data-tag="___react-data-table-allow-propagation___"
            >
              Conan the Barbarian 2
            </div>
          </div>
          <div
            className="sc-AxhCb sc-AxhUy sc-AxgMl gbbhfF rdt_TableCell"
            data-tag="___react-data-table-allow-propagation___"
            id="cell-2-2"
            role="cell"
          >
            <div
              data-tag="___react-data-table-allow-propagation___"
            >
              1984
            </div>
          </div>
        </div>
        <div
          className="sc-fzoLsD jekBnA rdt_TableRow"
          id="row-3"
          onClick={[Function]}
          onDoubleClick={[Function]}
          role="row"
          selected={false}
        >
          <div
            className="sc-AxhCb sc-AxhUy sc-AxgMl gbbhfF rdt_TableCell"
            data-tag="___react-data-table-allow-propagation___"
            id="cell-1-3"
            role="cell"
          >
            <div
              data-tag="___react-data-table-allow-propagation___"
            >
              Conan the Barbarian 3
            </div>
          </div>
          <div
            className="sc-AxhCb sc-AxhUy sc-AxgMl gbbhfF rdt_TableCell"
            data-tag="___react-data-table-allow-propagation___"
            id="cell-2-3"
            role="cell"
          >
            <div
              data-tag="___react-data-table-allow-propagation___"
            >
              1985
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`DataTable should render pagination 1`] = `
<div
  className="sc-fznxsB forRdO"
  style={Object {}}
>
  <div
    className="sc-fznWqX gnahTY"
  >
    <div
      className="sc-AxjAm gIMaKV rdt_Table"
      disabled={false}
      role="table"
    >
      <div
        className="sc-AxirZ eBuPmJ rdt_TableHead"
        role="rowgroup"
      >
        <div
          className="sc-AxiKw cTEQON rdt_TableHeadRow"
          disabled={false}
          role="row"
        >
          <div
            className="sc-AxhCb sc-AxhUy sc-fzplWN HWHoi rdt_TableCol"
          >
            <div
              className="sc-fznyAO cxtWCz rdt_TableCol_Sortable"
              id="column-title"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="button"
              tabIndex={0}
            >
              <div>
                Title
              </div>
            </div>
          </div>
          <div
            className="sc-AxhCb sc-AxhUy sc-fzplWN HWHoi rdt_TableCol"
          >
            <div
              className="sc-fznyAO cxtWCz rdt_TableCol_Sortable"
              id="column-year"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="button"
              tabIndex={0}
            >
              <div>
                Year
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="sc-fzpjYC ieTOoW rdt_TableBody"
        offset="250px"
        role="rowgroup"
      >
        <div
          className="sc-fzoLsD jekBnA rdt_TableRow"
          id="row-1"
          onClick={[Function]}
          onDoubleClick={[Function]}
          role="row"
          selected={false}
        >
          <div
            className="sc-AxhCb sc-AxhUy sc-AxgMl gbbhfF rdt_TableCell"
            data-tag="___react-data-table-allow-propagation___"
            id="cell-1-1"
            role="cell"
          >
            <div
              data-tag="___react-data-table-allow-propagation___"
            >
              Conan the Barbarian
            </div>
          </div>
          <div
            className="sc-AxhCb sc-AxhUy sc-AxgMl gbbhfF rdt_TableCell"
            data-tag="___react-data-table-allow-propagation___"
            id="cell-2-1"
            role="cell"
          >
            <div
              data-tag="___react-data-table-allow-propagation___"
            >
              1982
            </div>
          </div>
        </div>
        <div
          className="sc-fzoLsD jekBnA rdt_TableRow"
          id="row-2"
          onClick={[Function]}
          onDoubleClick={[Function]}
          role="row"
          selected={false}
        >
          <div
            className="sc-AxhCb sc-AxhUy sc-AxgMl gbbhfF rdt_TableCell"
            data-tag="___react-data-table-allow-propagation___"
            id="cell-1-2"
            role="cell"
          >
            <div
              data-tag="___react-data-table-allow-propagation___"
            >
              Conan the Barbarian 2
            </div>
          </div>
          <div
            className="sc-AxhCb sc-AxhUy sc-AxgMl gbbhfF rdt_TableCell"
            data-tag="___react-data-table-allow-propagation___"
            id="cell-2-2"
            role="cell"
          >
            <div
              data-tag="___react-data-table-allow-propagation___"
            >
              1984
            </div>
          </div>
        </div>
        <div
          className="sc-fzoLsD jekBnA rdt_TableRow"
          id="row-3"
          onClick={[Function]}
          onDoubleClick={[Function]}
          role="row"
          selected={false}
        >
          <div
            className="sc-AxhCb sc-AxhUy sc-AxgMl gbbhfF rdt_TableCell"
            data-tag="___react-data-table-allow-propagation___"
            id="cell-1-3"
            role="cell"
          >
            <div
              data-tag="___react-data-table-allow-propagation___"
            >
              Conan the Barbarian 3
            </div>
          </div>
          <div
            className="sc-AxhCb sc-AxhUy sc-AxgMl gbbhfF rdt_TableCell"
            data-tag="___react-data-table-allow-propagation___"
            id="cell-2-3"
            role="cell"
          >
            <div
              data-tag="___react-data-table-allow-propagation___"
            >
              1985
            </div>
          </div>
        </div>
        <div
          className="sc-fzoLsD jekBnA rdt_TableRow"
          id="row-4"
          onClick={[Function]}
          onDoubleClick={[Function]}
          role="row"
          selected={false}
        >
          <div
            className="sc-AxhCb sc-AxhUy sc-AxgMl gbbhfF rdt_TableCell"
            data-tag="___react-data-table-allow-propagation___"
            id="cell-1-4"
            role="cell"
          >
            <div
              data-tag="___react-data-table-allow-propagation___"
            >
              Conan the Barbarian
            </div>
          </div>
          <div
            className="sc-AxhCb sc-AxhUy sc-AxgMl gbbhfF rdt_TableCell"
            data-tag="___react-data-table-allow-propagation___"
            id="cell-2-4"
            role="cell"
          >
            <div
              data-tag="___react-data-table-allow-propagation___"
            >
              1982
            </div>
          </div>
        </div>
        <div
          className="sc-fzoLsD jekBnA rdt_TableRow"
          id="row-5"
          onClick={[Function]}
          onDoubleClick={[Function]}
          role="row"
          selected={false}
        >
          <div
            className="sc-AxhCb sc-AxhUy sc-AxgMl gbbhfF rdt_TableCell"
            data-tag="___react-data-table-allow-propagation___"
            id="cell-1-5"
            role="cell"
          >
            <div
              data-tag="___react-data-table-allow-propagation___"
            >
              Conan the Barbarian 2
            </div>
          </div>
          <div
            className="sc-AxhCb sc-AxhUy sc-AxgMl gbbhfF rdt_TableCell"
            data-tag="___react-data-table-allow-propagation___"
            id="cell-2-5"
            role="cell"
          >
            <div
              data-tag="___react-data-table-allow-propagation___"
            >
              1984
            </div>
          </div>
        </div>
        <div
          className="sc-fzoLsD jekBnA rdt_TableRow"
          id="row-6"
          onClick={[Function]}
          onDoubleClick={[Function]}
          role="row"
          selected={false}
        >
          <div
            className="sc-AxhCb sc-AxhUy sc-AxgMl gbbhfF rdt_TableCell"
            data-tag="___react-data-table-allow-propagation___"
            id="cell-1-6"
            role="cell"
          >
            <div
              data-tag="___react-data-table-allow-propagation___"
            >
              Conan the Barbarian 3
            </div>
          </div>
          <div
            className="sc-AxhCb sc-AxhUy sc-AxgMl gbbhfF rdt_TableCell"
            data-tag="___react-data-table-allow-propagation___"
            id="cell-2-6"
            role="cell"
          >
            <div
              data-tag="___react-data-table-allow-propagation___"
            >
              1985
            </div>
          </div>
        </div>
        <div
          className="sc-fzoLsD jekBnA rdt_TableRow"
          id="row-7"
          onClick={[Function]}
          onDoubleClick={[Function]}
          role="row"
          selected={false}
        >
          <div
            className="sc-AxhCb sc-AxhUy sc-AxgMl gbbhfF rdt_TableCell"
            data-tag="___react-data-table-allow-propagation___"
            id="cell-1-7"
            role="cell"
          >
            <div
              data-tag="___react-data-table-allow-propagation___"
            >
              Conan the Barbarian
            </div>
          </div>
          <div
            className="sc-AxhCb sc-AxhUy sc-AxgMl gbbhfF rdt_TableCell"
            data-tag="___react-data-table-allow-propagation___"
            id="cell-2-7"
            role="cell"
          >
            <div
              data-tag="___react-data-table-allow-propagation___"
            >
              1982
            </div>
          </div>
        </div>
        <div
          className="sc-fzoLsD jekBnA rdt_TableRow"
          id="row-8"
          onClick={[Function]}
          onDoubleClick={[Function]}
          role="row"
          selected={false}
        >
          <div
            className="sc-AxhCb sc-AxhUy sc-AxgMl gbbhfF rdt_TableCell"
            data-tag="___react-data-table-allow-propagation___"
            id="cell-1-8"
            role="cell"
          >
            <div
              data-tag="___react-data-table-allow-propagation___"
            >
              Conan the Barbarian 2
            </div>
          </div>
          <div
            className="sc-AxhCb sc-AxhUy sc-AxgMl gbbhfF rdt_TableCell"
            data-tag="___react-data-table-allow-propagation___"
            id="cell-2-8"
            role="cell"
          >
            <div
              data-tag="___react-data-table-allow-propagation___"
            >
              1984
            </div>
          </div>
        </div>
        <div
          className="sc-fzoLsD jekBnA rdt_TableRow"
          id="row-9"
          onClick={[Function]}
          onDoubleClick={[Function]}
          role="row"
          selected={false}
        >
          <div
            className="sc-AxhCb sc-AxhUy sc-AxgMl gbbhfF rdt_TableCell"
            data-tag="___react-data-table-allow-propagation___"
            id="cell-1-9"
            role="cell"
          >
            <div
              data-tag="___react-data-table-allow-propagation___"
            >
              Conan the Barbarian 3
            </div>
          </div>
          <div
            className="sc-AxhCb sc-AxhUy sc-AxgMl gbbhfF rdt_TableCell"
            data-tag="___react-data-table-allow-propagation___"
            id="cell-2-9"
            role="cell"
          >
            <div
              data-tag="___react-data-table-allow-propagation___"
            >
              1985
            </div>
          </div>
        </div>
        <div
          className="sc-fzoLsD jekBnA rdt_TableRow"
          id="row-10"
          onClick={[Function]}
          onDoubleClick={[Function]}
          role="row"
          selected={false}
        >
          <div
            className="sc-AxhCb sc-AxhUy sc-AxgMl gbbhfF rdt_TableCell"
            data-tag="___react-data-table-allow-propagation___"
            id="cell-1-10"
            role="cell"
          >
            <div
              data-tag="___react-data-table-allow-propagation___"
            >
              Conan the Barbarian
            </div>
          </div>
          <div
            className="sc-AxhCb sc-AxhUy sc-AxgMl gbbhfF rdt_TableCell"
            data-tag="___react-data-table-allow-propagation___"
            id="cell-2-10"
            role="cell"
          >
            <div
              data-tag="___react-data-table-allow-propagation___"
            >
              1982
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <nav
    className="sc-fzoNJl dGqwdB rdt_Pagination"
  >
    <span
      className="sc-fznxKY sc-fznWOq ncoBp"
    >
      Rows per page:
    </span>
    <div
      className="sc-fzoyTs gVljTM"
    >
      <select
        className="sc-fzqNqU gCcqTE"
        defaultValue={10}
        onChange={[Function]}
      >
        <option
          value={10}
        >
          10
        </option>
        <option
          value={15}
        >
          15
        </option>
        <option
          value={20}
        >
          20
        </option>
        <option
          value={25}
        >
          25
        </option>
        <option
          value={30}
        >
          30
        </option>
      </select>
      <svg
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </div>
    <span
      className="sc-fznxKY sc-fznMAR jMFmnk"
    >
      1-10 of 11
    </span>
    <div
      className="sc-fzpmMD ftIMtk"
    >
      <button
        aria-disabled={true}
        aria-label="First Page"
        className="sc-fzoXWK embYcA"
        disabled={true}
        id="pagination-first-page"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          height="24"
          role="presentation"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18.41 16.59L13.82 12l4.59-4.59L17 6l-6 6 6 6zM6 6h2v12H6z"
          />
          <path
            d="M24 24H0V0h24v24z"
            fill="none"
          />
        </svg>
      </button>
      <button
        aria-disabled={true}
        aria-label="Previous Page"
        className="sc-fzoXWK embYcA"
        disabled={true}
        id="pagination-previous-page"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          height="24"
          role="presentation"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
      </button>
      <button
        aria-disabled={false}
        aria-label="Next Page"
        className="sc-fzoXWK embYcA"
        disabled={false}
        id="pagination-next-page"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          height="24"
          role="presentation"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
      </button>
      <button
        aria-disabled={false}
        aria-label="Last Page"
        className="sc-fzoXWK embYcA"
        disabled={false}
        id="pagination-last-page"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          height="24"
          role="presentation"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5.59 7.41L10.18 12l-4.59 4.59L7 18l6-6-6-6zM16 6h2v12h-2z"
          />
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
        </svg>
      </button>
    </div>
  </nav>
</div>
`;
